model/buildlogger.go

```diff
@@ -266,6 +271,154 @@ func (l *Log) CloseLog(completedAt time.Time, exitCode int) error {
 
 }
 
+// Download returns a LogIterator which iterates lines of the given log. The
+// log should be populated and the environment should not be nil.
+func (l *Log) Download() (*LogIterator, error) {
```

Comment:
we should also benchmark this.

Reply:
it might also be good to set up an API so that we can get ranges? and maybe (optionally force serialization? It might be the case that buffering in this way isn't always good, and it'd be nice to be able to select "serial" vs "buffer everything" 

if there end up being a lot of files, this method might also hit open-file descriptor limits.